### PR TITLE
[nrunner] inject run.test_parameters option on Test params

### PR DESCRIPTION
**Pull request by willianrampazzo**
Signed-off-by: Willian Rampazzo <willianr@redhat.com>

### CODE REVIEW

**Comment by willianrampazzo** [comment]
I'll handle the exec-test similar problem in a different PR.

File: avocado/core/runners/avocado_instrumented.py [hunk 1]
```diff
@@ -104,12 +104,20 @@ def _run_avocado(runnable, queue):
...
 
             klass, method = klass_method.split('.', 1)
+
+            # inject the run.test_parameters option in the Test params
+            # this handles the -p command-line argument
+            test_parameters = {name: value for name, value in
+                               runnable.config.get('run.test_parameters', [])}
+            params = TreeNode().get_node("/", True)
+            params.value = test_parameters
+
```
Comment (beraldoleal) [inline avocado/core/runners/avocado_instrumented.py:114]: IIUC, `run.test_parameters` is a list of lists. Something like:

```python
"run.test_parameters": [["foo", "1"]]
```

So what do you think about this:

```python
params = TreeNode().get_node("/", True)
params.value = dict(runnable.config.get('run.test_parameters'))
```

Note: AFAICT, config.get() should return all the values, and the default value for `run.test_parameters` is already a empty list. So, when getting, the default `[]` wouldn't be necessary. But if that is not the case here, then you can keep the `[]`.

**Review by beraldoleal — COMMENTED** [review]
Hi @willianrampazzo this LGTM, just one small comment for you.

File: avocado/core/runners/avocado_instrumented.py (hunk 1 above)
Comment (willianrampazzo) [reply avocado/core/runners/avocado_instrumented.py:112]: Much better, thanks! I force-pushed the change.

**Review by beraldoleal — APPROVED** [review]
LGTM.

**Comment by willianrampazzo** [comment]
@beraldoleal, I had to add the default value back as some tests running with the Job API are failing.

**Review by clebergnu — APPROVED** [review]
LGTM, thanks!

**Pull request merged**